src/main/java/com/sogeti/SoSenseiPsyche/model/Feedback.java

```diff
@@ -9,6 +9,14 @@
 public class Feedback {
 
     public String getFeedback(List<Character> secretCode, List<Character> guessedCode) {
+
+        // actually this is to validate that the secretCode and the guessCode are always valid
+        if (secretCode == null || guessedCode == null) {
+            throw new IllegalArgumentException("Secret code or guessed code cannot be null.");
+        }
+        if (secretCode.size() != guessedCode.size()) {
+            throw new IllegalArgumentException("Secret code and guessed code must be of the same length.");
```

Comment:
The user doesn't know what is meant with "guessed code" and does not have any influence on the length of the secret code. So it would be more informative to say something like: "Please enter a code of four characters. The secret code consists of four characters."